### ISSUE TITLE
Nested env variables do not get parsed

### ISSUE THREAD

**Issue by aldo-roman** [cypress-io/cypress#4865]
**Current behavior:**
Cypress does not parse object env variables. Example:
```sh
// none of these gets parsed
export CYPRESS_REPORTER_OPTIONS=mochaFile=path/to/report.xml
export CYPRESS_REPORTER_OPTIONS='mochaFile=path/to/report.xml'
export CYPRESS_REPORTER_OPTIONS=mochaFile='path/to/report.xml'
export CYPRESS_REPORTER_OPTIONS={'mochaFile':'path/to/report.xml'}
```

This happens because the `getProcessEnvVars` function expects only string or boolean values:

https://github.com/cypress-io/cypress/blob/44c04160fdf266b74bdc3bd5817eb66d6d652dc0/packages/server/lib/config.coffee#L583-L588

https://github.com/cypress-io/cypress/blob/44c04160fdf266b74bdc3bd5817eb66d6d652dc0/packages/server/lib/util/coerce.js#L10-L18

**Expected behavior**
Parse nested env variables, similar to what is done for args
https://github.com/cypress-io/cypress/blob/680a2fccd29fb583f7d17a17056a9d15c2d26af9/packages/server/lib/util/args.js#L112

**Comment by GMchris** [comment]
I ran into this problem as well. I'm not sure what the idea behind it is, however I'm currently `JSON.stringify`-ing my objects to sidestep it. I'm curious whether this is intentional or not.

**Comment by jennifer-shehane** [comment]
Closing as duplicate of https://github.com/cypress-io/cypress/issues/1736

**Comment by jordandlaman** [comment]
Why was this closed as a duplicate of #1736 as this specifically pertains to how CYPRESS_* vars get parsed when setting CYPRESS_REPORTER_OPTIONS when #1736 pertains to deeply nested values in the plugins section?  Have tested this in 4.11.0 when #1736 was merged in and using CYPRESS_REPORTER_OPTIONS from the command is still a no go.

**Comment by jennifer-shehane** [comment]
For more complex configuration objects, you may want to consider passing a [JSON.stringified](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) object surrounded by single quotes.

```
CYPRESS_REPORTER_OPTIONS='{"mochaFile":"path/to/report.xml}'
```

`plugins/index.js`

```js
module.exports = (on, config) => {
  console.log(config) // see what all is in here!
  
  // return config
  return config
}
```
